configs/components/rubygem-puppet_forge.rb

```diff
@@ -1,6 +1,11 @@
 component 'rubygem-puppet_forge' do |pkg, settings, platform|
-  pkg.version '3.2.0'
-  pkg.md5sum '501d5f9f742007504d0d60ce6cf0c27f'
+  if settings[:ruby_version].to_f >= 3.2
+    pkg.version '6.0.0'
```

Comment:
Hi @danadoherty639 in situations where a gem is shared across multiple projects or branches, we generally define a new setting like `rubygem_<gem>_version` and have the setting default to the current version, for example `3.2.0`. Then in your project, do `settings[:rubygem_<gem>_version] = '6.0.0'` so that only your project pulls in the new version of the gem. For example, see https://github.com/puppetlabs/puppet-runtime/blob/master/configs/components/rubygem-concurrent-ruby.rb

However, if the `puppet_forge` is only used by the `pdk` project and there is only one active branch for `pdk`, then feel free to just change the version from `3.2.0` to `6.0.0` and update the checksum. A good way to check if your changes will only affect the pdk is to run from your feature branch:

```
bundle exec rake vanagon:component_diff -- --from upstream/main --to HEAD --project all --platform el-7-x86_64
```

A second issue is when updating a rubygem component, you may need to update its gem dependencies. For example, version 6.0.0 of https://rubygems.org/gems/puppet_forge/versions/6.0.0 has runtime dependencies:

[faraday](https://rubygems.org/gems/faraday) ~> 2.0
[faraday-follow_redirects](https://rubygems.org/gems/faraday-follow_redirects) ~> 0.3.0
[minitar](https://rubygems.org/gems/minitar) ~> 1.0, >= 1.0.2
[semantic_puppet](https://rubygems.org/gems/semantic_puppet) ~> 1.0

But 3.2.0 depends on 

[faraday](https://rubygems.org/gems/faraday) ~> 1.3
[faraday_middleware](https://rubygems.org/gems/faraday_middleware) ~> 1.0
[minitar](https://rubygems.org/gems/minitar) >= 0
[semantic_puppet](https://rubygems.org/gems/semantic_puppet) ~> 1.0

So you'll need to make sure the `faraday`, etc rubygem components in this repo satisfy the constraints for `puppet_forge` 6.0.0. In particular, the `minitar` component in this repo is not compatible with the new `puppet_forge` 6.0.0 gem

https://github.com/puppetlabs/puppet-runtime/blob/ecf1c5a3f158d4c5ac0fb70bce11fcb50abf2446/configs/components/rubygem-minitar.rb#L2

Since `minitar` is shared with the agent runtime and others, you'll want to create a `rubygems_minitar_version` setting in the `rubygems_minitar.rb` component and update the pdk to override the `rubygems_minitar_version`. And make sure your dependencies are added to the project in reverse dependency order like:

```
proj.component 'rubygem-minitar'
proj.component 'rubygem-faraday'
...
proj.component 'rubygem-puppet_forge'
```

Reply:
Hi @joshcooper, this PR is updated now following your instructions, let me know if there is anything else I'm missing thanks